pkg/engine/overlayCondition.go

```diff
@@ -1,174 +1,190 @@
 package engine
 
 import (
+	"fmt"
 	"reflect"
+	"strconv"
 
 	"github.com/golang/glog"
 	"github.com/nirmata/kyverno/pkg/engine/anchor"
 )
 
-func meetConditions(resource, overlay interface{}) bool {
+func meetConditions(resource, overlay interface{}) (string, error) {
+	return checkConditions(resource, overlay, "/")
+}
+
+// resource and overlay should be the same type
+func checkConditions(resource, overlay interface{}, path string) (string, error) {
 	// overlay has no anchor, return true
 	if !hasNestedAnchors(overlay) {
-		return true
+		return "", nil
 	}
 
 	// resource item exists but has different type
 	// return false if anchor exists in overlay
 	// conditon never be true in this case
 	if reflect.TypeOf(resource) != reflect.TypeOf(overlay) {
 		if hasNestedAnchors(overlay) {
-			glog.V(3).Infof("Found anchor on different types of element: overlay %T, resource %T\nSkip processing overlay.", overlay, resource)
-			return false
+			glog.V(4).Infof("Found anchor on different types of element at path %s: overlay %T, resource %T", path, overlay, resource)
+			return path, fmt.Errorf("Found anchor on different types of element at path %s: overlay %T %v, resource %T %v", path, overlay, overlay, resource, resource)
+
 		}
-		return true
+		return "", nil
 	}
 
-	return checkConditions(resource, overlay)
-}
-
-func checkConditions(resource, overlay interface{}) bool {
 	switch typedOverlay := overlay.(type) {
 	case map[string]interface{}:
 		typedResource := resource.(map[string]interface{})
-		return checkConditionOnMap(typedResource, typedOverlay)
+		return checkConditionOnMap(typedResource, typedOverlay, path)
 	case []interface{}:
 		typedResource := resource.([]interface{})
-		return checkConditionOnArray(typedResource, typedOverlay)
+		return checkConditionOnArray(typedResource, typedOverlay, path)
 	default:
-		return true
+		// anchor on non map/array is invalid:
```

Comment:
add error log for default, as the comment refers to an invalid scenario

Reply:
Added.